### DISCUSSION TITLE
How do system requirements work?

### DISCUSSION THREAD

**Discussion by twrightsman** [prefix-dev/pixi#301] [Q&A] [answered]
I'm a bit confused on how the `system-requirements` section works in `pixi.toml`.

When I specify:

```toml
[system-requirements]
cuda = "11"
```

I assume this means CUDA 11 is required by this environment? How could I specify any version greater than 11? It seems `pixi` requires an actual version number in this section.

**Comment by baszalmstra** [comment, marked as answer]
System requirements basically configure the reference machine for your project. So any system that has at least those specs will work for your project. In essence, it sets the virtual package `__cuda` to `__cuda=11=0` when you solve an environment. So this means that if a package requires `__cuda >=11.0` it is installable. If a package requires `__cuda>=12` it wont be installable.

When you run any task with pixi, it will check if the specs from your system are at least equal to or higher than the system requirements specified in your pixi.toml. So with the specification you mentioned above, if I don't have cuda or a version lower than 11, I wont be able to run your project. If I have cuda 11 or higher, I'm guaranteed that your project will run.

**Reply by twrightsman** [reply]
Got it, that makes sense. So in the current design of system requirements it is not possible to set a maximum or set of versions of virtual packages that won't work? e.g. this environment works on CUDA 11, except 11.3, but not CUDA 12.

**Reply by baszalmstra** [reply]
Not as of right now. This is something that we are looking into though. I think its related to https://github.com/prefix-dev/pixi/issues/239 and the answers there.

**Comment by twrightsman** [comment]
Link to the new documentation on system requirements for future readers: https://prefix.dev/docs/pixi/configuration#the-system-requirements-table
